src/databricks/labs/lakebridge/transpiler/repository.py

```diff
@@ -56,45 +73,91 @@ def transpilers_path(self) -> Path:
         return self._labs_path / "remorph-transpilers"
 
     def get_installed_version(self, product_name: str) -> str | None:
+        """
+        Obtain the version of an installed transpiler.
+
+        Args:
+          product_name: The product name of the transpiler whose version is sought.
+        Returns:
+          The version of the transpiler if it is installed, or None otherwise.
+        """
         # Warning: product_name here (eg. 'morpheus') and transpiler_name elsewhere (eg. Morpheus) are not the same!
         product_path = self.transpilers_path() / product_name
         current_version_path = product_path / "state" / "version.json"
-        if not current_version_path.exists():
+        try:
+            text = current_version_path.read_text("utf-8")
+        except FileNotFoundError:
             return None
-        text = current_version_path.read_text("utf-8")
```

Comment:
Generally when opening files it's an anti-pattern to try and predict whether it can actually be opened. Instead it's preferable to simply try opening it and handle failure.